webapp/src/context/AuthContext.tsx

```diff
@@ -60,177 +76,146 @@ const AppAuthProvider = ({ children }: { children: React.ReactNode }) => {
-        setUI("logout");
-        return;
-      }
+        if (state.isAuthenticated) {
+          setAppState(AppState.Authenticating);
+          await setupAuthenticatedUser();
 
-      if (!localStorage.getItem("people-app-redirect-url")) {
-        localStorage.setItem(
-          "people-app-redirect-url",
-          window.location.href.replace(window.location.origin, "")
-        );
-      }
-
-      if (!asg.isAuthenticated) {
-        await signIn();
-        return;
-      }
+          if (mounted) setAppState(AppState.Authenticated);
+          
+        } else {
+          const silentSignInSuccess = await trySignInSilently();
 
-      try {
-        const [basic, idToken, decoded] = await Promise.all([
-          getBasicUserInfo(),
-          getIDToken(),
-          getDecodedIDToken(),
-        ]);
-        if (cancelled) return;
-
-        dispatch(
-          setUserAuthData({ userInfo: basic, idToken, decodedIdToken: decoded })
-        );
-        new APIService(idToken, refreshToken);
-
-        await dispatch(getUserInfo()).unwrap();
-        await dispatch(loadPrivileges()).unwrap();
-
-        if (!cancelled) setUI("active");
-      } catch (err: unknown) {
-        console.error("Failed to logging in :", err);
-
-        const status = (err as any)?.response?.status ?? (err as any)?.status;
-        if (status === 401 || status === 403) {
-          if (!cancelled) {
-            localStorage.removeItem("people-app-state");
-            await signIn();
-          }
-          return;
+          if (mounted) setAppState(silentSignInSuccess ? AppState.Authenticating : AppState.Unauthenticated);
         }
-
-        try {
-          const stillAuth = await isAuthenticated();
-          if (!stillAuth) {
-            await signIn();
-            return;
-          }
-        } catch {
-          await signIn();
-          return;
+      } catch (err) {
+        if (mounted) {
+          dispatch(setAuthError());
         }
-
-        if (!cancelled) setUI("error");
       }
     };
 
-    bootstrap();
+    initializeAuth();
+
     return () => {
-      cancelled = true;
+      mounted = false;
     };
-  }, [
-    asg.isAuthenticated,
-    asg.isLoading,
-    signIn,
-    dispatch,
-    getBasicUserInfo,
-    getIDToken,
-    getDecodedIDToken,
-    isAuthenticated,
-  ]);
-
-  useEffect(() => {
-    if (ui === "active" && !asg.isAuthenticated) {
-      signIn();
-    }
-  }, [ui, asg.isAuthenticated, signIn]);
+  }, [state.isAuthenticated, state.isLoading]);
+
+  const refreshToken = () => {
+    return new Promise<{ idToken: string }>(async (resolve) => {
+      const userIsAuthenticated = await isAuthenticated();
+      if (userIsAuthenticated) {
+        resolve({ idToken: await getIDToken() });
+      } else {
+        refreshAccessToken()
+          .then(async () => {
+            const idToken = await getIDToken();
+            resolve({ idToken: idToken });
```

Comment:
```suggestion
      if (userIsAuthenticated) {
        resolve({ idToken: await getAccessToken() });
      } else {
        refreshAccessToken()
          .then(async () => {
            const idToken = await getAccessToken();
            resolve({ idToken: idToken });
```
We use idToken only for authentication. For authorization accessToken should be used. Please check whether this works.